File: 17_FinalProject/Coupon System/CouponSystem2_Spring/src/main/java/JavaProject/CouponSystem2_Spring/Beans/Coupon.java
package JavaProject.CouponSystem2_Spring.Beans;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;
import java.time.LocalDate;
import java.util.List;
import java.util.Set;

/**
 * Bean: Coupon - used to define a coupon listed in the system
 */
@Entity
@Table(name = "coupons")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Coupon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "company_id",
            nullable = false)
    private Integer companyId;

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "category_id",
            nullable = false)
    private Category category;

    @Column(name = "title",
            length = 50,
            unique = true)
    @Length(max = 50)
    private String title;

    @Column(name = "description",
            length = 70)
    @Length(max = 70)
    private String description;

    @Column(name = "start_date")
    private LocalDate start_date;

    @Column(name = "end_date")
    private LocalDate end_date;

    @Column(name = "amount")
    private int amount;

    @Column(scale = 2,
            name = "price")
    private double price;

    @Column(name = "image")
    private String image;

}

File: 17_FinalProject/Coupon System/CouponSystem3_FrontEnd/couponswebsite/src/Models/Coupon.ts

import { CouponCategory } from "./CouponCategory";

export class Coupon{
    public id:number;
    public companyId:number;
    public category:CouponCategory;
    public title:string;
    public description:string;
    public start_date:string;
    public end_date:string;
    public amount:number;
    public price:number;
    public image:string;

    constructor(id:number,companyId:number,category:CouponCategory,title:string,description:string,
        start_date:string,end_date:string,amount:number,price:number,image:string){
            this.id=id;
            this.companyId=companyId;
            this.category=category;
            this.title=title;
            this.description=description;
            this.start_date=start_date;
            this.end_date=end_date;
            this.amount=amount;
            this.price=price;
            this.image=image;
    }

}